doc/tunnel/SAI-Proposal-EVPN-Multihoming.md

```diff
@@ -0,0 +1,616 @@
+            attrs.push_back(attr);
+
+            attr.id = SAI_FDB_ENTRY_ATTR_ALLOW_MAC_MOVE;
+            attr.value.booldata = true;
+            attrs.push_back(attr);
+
+            status = sai_fdb_api->create_fdb_entry(&fdb_entry, (uint32_t)attrs.size(), attrs.data());
+```
+
+## 4.2 Split Horizon workflow
+
+  When Tunnel objects of peer mode type P2P are created, the isolation group objects can be re-used 
+  to achieve the split horizon functionality and do not need the attributes being introduced as part of this
+  PR. It is being elaborated here for completeness.
+
+![EVPN Multihoming](figures/sai_evpnmh_splithorizon.png "Figure 1: Split Horizon")
+__Figure 2: Split Horizon Flow__
+
+  At VTEP1 the following SAI objects with sub types are created.
+
+  - SAI_OBJECT_TYPE_TUNNEL with peer mode as SAI_TUNNEL_PEER_MODE_P2P, tnl_oid_2-4 created for tunnels towards the peer multihoming VTEP2-4.
+      Please refer to sec 4.1.
+  - SAI_OBJECT_TYPE_BRIDGE_PORT of type SAI_BRIDGE_PORT_TYPE_TUNNEL, bp_tnl_oid_2-4 corresponding to tunnels as above. 
+  - SAI_OBJECT_TYPE_BRIDGE_PORT of type SAI_BRIDGE_PORT_TYPE_PORT, bp_lag_oid_1-2 corresponding to client side LAGs 1,2.
+  - SAI_OBJECT_TYPE_ISOLATION_GROUP of type SAI_ISOLATION_GROUP_TYPE_BRIDGE_PORT, isogrp_oid_2-4 corresponding to bp_tnl_oid_2-4 as above.
+  - bp_tnl_oid_2-4 have SAI_BRIDGE_PORT_ATTR_ISOLATION_GROUP set as isogrp_oid_2-4 respectively.
+
+```
+    sai_attribute_t attr;
+    attr.value.s32 = SAI_ISOLATION_GROUP_TYPE_BRIDGE_PORT;
+    status = sai_isolation_group_api->create_isolation_group(&isogrp_oid_2, gSwitchId, 1, &attr);
+
+    attr.id = SAI_BRIDGE_PORT_ATTR_ISOLATION_GROUP;
+    attr.value.oid = isogrp_oid_2;
+    status = sai_bridge_api->set_bridge_port_attribute(bp_tnl_oid_2, &attr);
+ 
+    /* repeat for isogrp and bp_tnl_oid 3,4 */
+    ............
+    ............
+```
+
+  - Isolation Group member objects
+    - SAI_OBJECT_TYPE_ISOLATION_GROUP_MEMBER isogrp_mbr_oid_22 corresponding to isogrp_oid_2+bp_lag_oid_2 ( Traffic from VTEP-2 will be dropped on LAG-2 )
+    - SAI_OBJECT_TYPE_ISOLATION_GROUP_MEMBER isogrp_mbr_oid_32 corresponding to isogrp_oid_3+bp_lag_oid_2 ( Traffic from VTEP-3 will be dropped on LAG-2 )
+    - SAI_OBJECT_TYPE_ISOLATION_GROUP_MEMBER isogrp_mbr_oid_41 corresponding to isogrp_oid_4+bp_lag_oid_1 ( Traffic from VTEP-4 will be dropped on LAG-1 )
+    - SAI_OBJECT_TYPE_ISOLATION_GROUP_MEMBER isogrp_mbr_oid_42 corresponding to isogrp_oid_4+bp_lag_oid_2 ( Traffic from VTEP-4 will be dropped on LAG-2 )
+
+```
+        sai_object_id_t isogrp_mbr_oid_22 = SAI_NULL_OBJECT_ID;
+        sai_attribute_t mem_attr[2];
+        sai_status_t status = SAI_STATUS_SUCCESS;
+
+        mem_attr[0].id = SAI_ISOLATION_GROUP_MEMBER_ATTR_ISOLATION_GROUP_ID;
+        mem_attr[0].value.oid = isogrp_oid_2;
+        mem_attr[1].id = SAI_ISOLATION_GROUP_MEMBER_ATTR_ISOLATION_OBJECT;
+        mem_attr[1].value.oid = bp_po_oid_2;

```

Comment:
What is bp_po_oid_2 ? No other reference to this.

Reply:
Can attaching the isolation group also be covered in the workflow.

Reply:
Can workflow for realizing local-bias be included ?

Reply:
bp_po_oid_2 should be renamed as bp_lag_oid_2. Will do the change.

Following lines show isolation group being attached to the bridgeport. 
505     attr.id = SAI_BRIDGE_PORT_ATTR_ISOLATION_GROUP;
506     attr.value.oid = isogrp_oid_2;
507     status = sai_bridge_api->set_bridge_port_attribute(bp_tnl_oid_2, &attr);

Local bias is basically MAC being HW learnt locally or MAC learnt over BGP from the peers being made to point to the local ES. These are existing constructs and fairly basic and hence did not cover them.

Reply:
@srj102 , regarding local bias, Assume this scenario.  In section "4.3 DF workflow". 
a) BUM traffic originates from a single homed server (say HOST-3) connected to VTEP-4.
b) This BUM packet reaches VTEP-3, which is the DF-node. It should not forward the packet to HOST-2.
c) At VTEP-3, this filtering should be based on 'ES and SIP of VTEP-4'.
How is this expected to be realized in your proposal ? I understand you are not considering P2MP now and the split horizon explained in 4.2 is expected to take effect. But is this proposal future extensible to support P2MP later ?